Implement configurable default auto-type delay (fixes #699)

hifi opened the pull request:
Adds both solutions presented in issue #699.

## Description
All platform dependant hardcoded auto-type delays were removed in favor of a configurable default. UI was added to configure the delay which defaults to 25ms as specified.

## Motivation and context
Issue #699.

## How has this been tested?
Only on Linux with three different auto-type entries:

1. Default, no custom delay (will use the configured default)
2. Sequence with {DELAY=0} -> will override the default
3. Sequence with {DELAY=200} -> will override the default

Tested the default at 0 ms, 25 ms and 500 ms. All work and are overridable by the sequence {DELAY=X} option.

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/106598/27646980-d1d08254-5c32-11e7-8944-c80e60863085.png)

## Types of changes
- ✅ Bug fix (non-breaking change which fixes an issue)
- ✅ New feature (non-breaking change which adds functionality)

## Checklist:
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**

phoerious commented on src/gui/SettingsWidgetGeneral.ui:
```diff
@@ -351,6 +351,58 @@
+          </widget>
+         </item>
+         <item>
+          <widget class="QSpinBox" name="autoTypeDelaySpinBox">
```
Please use a proper grid layout.

hifi replied:
Right, the global auto-type shortcut was already hacked together like this so I just continued the tradition. Just to clarify: you want me to wrap both of them in a grid like "Timeouts" is in "Security"?

phoerious replied:
It doesn't have one because it's the only option like that on this tab. But as soon as you have two or more, you need a grid t align them. So yes, please wrap those two options in a grid.

phoerious commented:
After addressing my review change request, please replay your changes on top of release/2.2.1 instead of develop (you can re-use the same branch by force-pushing, so we don't need a new PR).

hifi commented:
Changed to grid layout and rebased against release/2.2.1.

![image](https://user-images.githubusercontent.com/106598/27690715-c9c57336-5cea-11e7-8580-fd8701f46b22.png)

phoerious commented:
Thanks for your adjustments. Only thing missing: the two labels need to be right-aligned.

hifi commented:
Sorry, my oversight. Fixed now.

![image](https://user-images.githubusercontent.com/106598/27702431-bc1b7114-5d0c-11e7-9748-c54d660f474e.png)

hifi commented:
![image](https://user-images.githubusercontent.com/106598/27702797-e8407536-5d0d-11e7-831a-565340c1675c.png)

adatum commented:
Any chance of another release soon with the many changes committed and merged so far? In particular, this feature of a configurable delay (or no delay, as in versions prior to v2.2.0) has a big impact on usability with long passwords.
